src/hls-adapter.js

```diff
@@ -678,9 +699,20 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         HlsAdapter._logger.error("cannot recover, last media error recovery failed");
       }
     }
+    this._videoElement.addEventListener(EventType.LOADED_METADATA, this._onRecoveredCallback);
```

Comment:
@odedhutzler you would want to this only `if (recover) { ... }`.

Actually, because we only add the listener after we call `_recoverDecodingError` or `_recoverSwapAudioCodec` there's a chance that this might not even pick up, so better return this to be inlined twice, above and before each call to the recover handlers